xarray/core/missing.py

```diff
@@ -181,9 +181,9 @@ def get_clean_interp_index(arr, dim, method, use_coordinate=True, **kwargs):
                 # The division can change the nearest-neighbour
                 # when compared to pandas (which does not divide).
                 # Let's keep that compatitibility
-                index = (index - index.min())
+                index = (index - index.mean())
                 if len(index) > 1:
-                    index /= index.std()
+                    index /= np.max(np.abs(index))
```

Comment:
This still needs a fix to account for possibly dividing by zero.

Also, I don't think you need abs() here because subtracting by mean() already centers the data.

Reply:
The `abs()` makes sure that it's always within [-1, 1] right? e.g. if `index = [-2, 0, 0.9]` and as long as `index` is not all zeroes, it should always work? :thinking: 
Sorry, I feel like I'm missing something.

Reply:
Yes, this would work fine unless the array is all zeros/constant.

It still might be worth considering that possibility, though at this point I'm pretty sure we are just refining total edge case behavior.

Maybe:
```
min_value = index.min()
max_value = index.max()
range_ = max_value - min_value
if range_:
    midpoint = (min_value + max_value) / 2
    index = index - midpoint
    index /= range_
```

Reply:
The other option is to not really worry about this at all, besides silencing the warning. float64 is about the best we can do as far as precision, and it's a simple fact that you cannot represent every int64 value exactly in float64. In most cases this isn't a problem.

Reply:
The people doing lightning research probably don't think that scenario is all that unusual (don't know for sure, I just remember a conversation with a lightning researcher when he complained about matplotlib's lack of support for nanosecond tickers).

Reply:
> The other option is to not really worry about this at all, besides silencing the warning. float64 is about the best we can do as far as precision, and it's a simple fact that you cannot represent every int64 value exactly in float64. In most cases this isn't a problem.

Ya, so the choice is between using `filterwarnings` or rescaling to avoid the warning. I have a mild preference for rescaling because it "solves" the problem. Hopefully, the lightning research people will chime in / send in a PR if this is actually an issue.

Reply:
Let's just silence the warning here. I don't think rescaling actually fixes anything, and it's less efficient than silencing it.